bookkeeper-server/src/main/java/org/apache/bookkeeper/util/collections/ConcurrentLongHashMap.java

```diff
@@ -377,6 +377,16 @@ private V remove(long key, Object value, int keyHash) {
                             if (nextValueInArray == EmptyValue) {
                                 values[bucket] = (V) EmptyValue;
                                 --usedBuckets;
+
+                                // Cleanup all the buckets that were in `DeletedValue` state,
```

Comment:
It looks like we can create a common private method to reduce code duplication in this class
Smaller methods are better in java

Reply:
OK

Reply:
Fixed